### DISCUSSION TITLE
Memory consumption with pnpm

### DISCUSSION THREAD

**Discussion by bvincent** [pnpm/pnpm#8266] [Q&A] [answered]
Hello,

I actually have to make installation with pnpm on a docker container which is limited in term of memory. The container stops with Out Of Memory message during pnpm install. Unfortunately, I cannot share the project setup.

- Is there a way to decrease memory used by pnpm during installation even if that takes much time to install ?
- Is there an (out of box) way to troubleshot memory usage by pnpm ?

Regards,

**Comment by zkochan** [comment]
Memory consumption is much lower when there is an up-to-date lockfile. Why isn't your lockfile up to date? Don't you commit the lockfile to the repository?

**Comment by bvincent** [comment]
The lock file is delivered into the container (buildpack) but, as the monorepo workspace is composed by multiple apps, not all packages of the monorepo are delivered. That could be considered as outdated lock file and leads to higher memory consumption ?

**Reply by zkochan** [reply, marked as answer]
maybe. You can try to use `pnpm deploy` instead. Outside the container. Then just copy the results to the container.

**Reply by bvincent** [reply]
That is good solution to limit memory consumption when deploying into container, thanks for helping. Unfortunately, our module is too big to be archive quick before pushing.
